### DISCUSSION TITLE
What are the possible reasons causing "no crossing of 0.5 detected"?

### DISCUSSION THREAD

**Discussion by QingfangLiu** [cvnlab/GLMsingle#62] [Q&A]
Hi there, I ran into the error of "no crossing of 0.5 detected" when I'm running the GLMsingle code on **only one** subject. I know it's caused by the "findtailthreshold" function, but I don't see why it only occurs on one of all subjects. Both data and design of this subject were structured similarly to other subjects, and the intermediate variable "onoffR2" also looks similar to what's in other subjects. Thanks.

**Comment by kendrickkay** [comment]
Can you please send a screenshot of the onoffR2 figure?  Also, I assume you are "git pull" up to date?  Better yet, if you can send the onoffR2 matrix of values, that would be helpful.  Also, just to confirm the newest code simply issues a warning (not an error/crash), correct?

**Reply by QingfangLiu** [reply]
Oh I actually wasn't using the latest version (and thanks so much for the reminder!), then after updating the toolbox, the problem disappeared in that subject. 

However, I still get warnings in some of my other subjects' data like the following. 

Warning: Failed to converge in 100 iterations during replicate 1 for
gmdistribution with 2 components 
> In gmcluster (line 204)
In gmdistribution.fit (line 102)
In fitgmdist (line 135)
In findtailthreshold (line 45)
In GLMestimatesingletrial (line 572)

Here is a screenshot of the onoffR2 distribution from the subject's data with warnings (also data attached).

![onoffR2](https://user-images.githubusercontent.com/26177439/167714218-098c52c3-80f3-46f3-b14b-d2ae732385a1.jpg)

[onoffR2.mat.zip](https://github.com/cvnlab/GLMsingle/files/8664741/onoffR2.mat.zip)

**Comment by kendrickkay** [comment]
I tried findtailthreshold on the file you sent.  The results look fine, even though there is a warning (notice the threshold is reasonable).  However, the values seem quite low.  The R2 values are nominally in a range like 0 to 100 (percent).  Not sure why your values are very low?

![image](https://user-images.githubusercontent.com/2728338/167860161-ebfb375a-84f5-4811-a741-eaa8ae3757c8.png)

**Reply by QingfangLiu** [reply]
Thanks so much for looking into this for me and sorry about my late response. I tried different ways to set up my model (by changing the durations, ROIs), but all the onoffR2 values were pretty low (all below 1). Does that indicate the onoff models weren't fitted well to the data? In case it's relevant, I'm most interested in the midbrain signals. I interpolated my neural data from TR=2s to TR=1s. 

Do you have any suggestions for me to conduct some diagnostic analysis? Thank you.

**Comment by kendrickkay** [comment]
One possibility is that the heuristic idea that an "ON-OFF" model would be a reasonable approximate way to model your time series data might be a bad heuristic.  Like, can we imagine that all of your conditions/trials generally evoke a uniformly positive (or negative) response from a given voxel?    It might not be an actual problem, though, since the ONOFF model is relaly just to help with some voxel selection.

In terms of diagnostics, that's a larger question involving looking at your MRI data quality ... not sure there is a simple answer here.  Eyebaling/visualizing your images and double-checking your design matrix is probably your best bet.

**Reply by QingfangLiu** [reply]
I see. I will look into more details on my data and design matrix. Thanks again for all your helpful suggestions! Much appreciated.
